--- etc/default/grub	2011-05-03 21:14:58.000000000 +0700
+++ /etc/default/grub	2011-05-29 15:38:55.742197013 +0700
@@ -1,14 +1,14 @@
-GRUB_DEFAULT=0
-GRUB_TIMEOUT=5
+GRUB_DEFAULT=saved
+GRUB_TIMEOUT=3
 GRUB_DISTRIBUTOR="Arch Linux"
-GRUB_CMDLINE_LINUX_DEFAULT="quiet"
+GRUB_CMDLINE_LINUX_DEFAULT="quiet init=/bin/systemd"
 GRUB_CMDLINE_LINUX=""
 
 # Preload both GPT and MBR modules so that they are not missed
 GRUB_PRELOAD_MODULES="part_gpt part_msdos"
 
 # Uncomment to enable Hidden Menu, and optionally hide the timeout count
-#GRUB_HIDDEN_TIMEOUT=5
+GRUB_HIDDEN_TIMEOUT=1
 #GRUB_HIDDEN_TIMEOUT_QUIET=true
 
 # Uncomment to use basic console
