--- /etc/pacman.conf	2011-03-17 01:40:11.000000000 +0700
+++ /etc/pacman.conf	2011-03-18 11:11:58.710310432 +0700
@@ -29,11 +29,11 @@
 #NoExtract   =
 
 # Misc options (all disabled by default)
-#UseSyslog
-#ShowSize
-#UseDelta
+UseSyslog
+ShowSize
+UseDelta
 #TotalDownload
-#CheckSpace
+CheckSpace
 
 #
 # REPOSITORIES
@@ -58,8 +58,19 @@
 # repo name header and Include lines. You can add preferred servers immediately
 # after the header, and they will be used before the default mirrors.
 
-#[testing]
-#Include = /etc/pacman.d/mirrorlist
+[kde-snapshots]
+Server = http://arch.morrigan.ch/kde-snapshots/os/i686
+[kde-snapshots-extra]
+Server = http://arch.morrigan.ch/kde-snapshots-extra/os/i686
+
+[radeon]
+Server = http://spiralinear.org/perry3d/i686/
+
+[testing]
+Include = /etc/pacman.d/mirrorlist
 
 [core]
 Include = /etc/pacman.d/mirrorlist
@@ -67,8 +78,8 @@
 [extra]
 Include = /etc/pacman.d/mirrorlist
 
-#[community-testing]
-#Include = /etc/pacman.d/mirrorlist
+[community-testing]
+Include = /etc/pacman.d/mirrorlist
 
 [community]
 Include = /etc/pacman.d/mirrorlist
